api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -99,6 +99,10 @@ message TcpProxy {
     // Neither ``:-prefixed`` pseudo-headers nor the Host: header can be overridden.
     repeated config.core.v3.HeaderValueOption headers_to_add = 3
         [(validate.rules).repeated = {max_items: 1000}];
+
+    // Emit response headers to the downstream info filter state for consumption
+    // by the network filters. The filter state key is ``envoy.tcp_proxy.tunnel_response_headers``.
+    bool emit_response_headers = 4;
```

Comment:
I'd lean towards something akin to 
save_connect_headers but I'll defer to api shephards on this one :-)

Reply:
@markdroth For API review and deciding on naming.

Reply:
How about calling this `propagate_response_headers`?

Reply:
Renamed.